Framework/Geometry/src/Instrument.cpp

```diff
@@ -1014,7 +1014,18 @@ void Instrument::saveNexus(::NeXus::File *file,
 
   // Now the parameter map, as a NXnote via its saveNexus method
   if (isParametrized()) {
-    const Geometry::ParameterMap &params = *getParameterMap();
+    Geometry::ParameterMap params(*getParameterMap());
+    // Masking is in DetectorInfo. Insert it into ParameterMap so it is saved
+    // alongside other parameters.
+    if (m_detectorInfo) {
```

Comment:
That would be an temporary option as long as we have `API::DetectorInfo`, but I think `Beamline::DetectorInfo` should stay independent of the `Geometry` module and anything related to the instrument legacy?

Furthermore, the API of `DetectorInfo` is quite visible on the client side, so adding methods that should never be used is confusing. Of course that could be solved by declaring `Geometry::Instrument` as a `friend`, but in a sense that would add a cyclic inter-module dependency?